src/main/java/receiver/StreamProcessor.java

```diff
@@ -0,0 +1,18 @@
+package receiver;
+
+public class StreamProcessor implements IDataProcessor{
+	
+	private static final String DELIMITER = " ";
+	private IntegerStack tempStack = new IntegerStack(5, "Temperature");
```

Comment:
A stack is last-in, first-out. Is it really a stack you mean, or a sliding window?